File: app/src/main/java/com/ocse/androidbaselib/model/BaseModel.java
package com.ocse.androidbaselib.model;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;

import com.google.gson.Gson;
import com.ocse.androidbaselib.bean.UserBean;
import com.ocse.androidbaselib.bean.VersionBean;
import com.ocse.androidbaselib.retrofit.ApiRetrofit;
import com.ocse.baseandroid.base.BaseViewModel;
import com.ocse.baseandroid.retrofit.base.BaseObserver;
import com.ocse.baseandroid.utils.InstallApkUtils;
import com.ocse.baseandroid.utils.SharePerferenceUtil;


/**
 * @author hujiayi
 */
public class BaseModel extends BaseViewModel {

    public MutableLiveData<UserBean> userMutableLiveData = get(UserBean.class);

    public void getUser() {
        ApiRetrofit.Companion.getInstacne().login("admin", "123456").subscribe(new BaseObserver<UserBean>(getCompositeDisposable()) {
            @Override
            public void _onNext(@NonNull UserBean entity) {
                userMutableLiveData.postValue(entity);
                SharePerferenceUtil.INSTANCE.saveString("token",entity.getAccess_token());
//                getVersion();
            }
        });

    }
//    public void getVersion() {
//        ApiRetrofit.Companion.getInstacne().getversion().subscribe(new BaseObserver<VersionBean>(getCompositeDisposable()) {
//            @Override
//            public void _onNext(@NonNull VersionBean entity) {
//                Log.e("hu",new Gson().toJson(entity));
//                InstallApkUtils.Companion.checkVersion("http://10.81.108.65:8090/mobile/downloadFj?path="+entity.getData().getLJ(),"1.apk",false);
//            }
//        });
//
//    }
}

File: app/src/main/java/com/ocse/androidbaselib/model/BaseModel.kt
package com.ocse.androidbaselib.model

import androidx.lifecycle.MutableLiveData
import com.ocse.androidbaselib.bean.UserBean
import com.ocse.androidbaselib.retrofit.ApiRetrofit.Companion.instance
import com.ocse.baseandroid.base.BaseViewModel
import com.ocse.baseandroid.net.base.BaseObserver

/**
 * @author hujiayi
 */
class BaseModel : BaseViewModel() {
    val userMutableLiveData by lazy { MutableLiveData<UserBean>() }
    val ss by lazy { MutableLiveData<UserBean>() }

    fun user() {
       instance.login("admin", "123456")
            .subscribe(object : BaseObserver<UserBean>(compositeDisposable) {
                override fun onError(e: Throwable) {
                    super.onError(e)
                    userMutableLiveData.postValue(null)
                }

                override fun _onNext(entity: UserBean) {
                    userMutableLiveData.postValue(entity)
                }
//                    override fun _onNext(entity: UserBean) {
//                        userMutableLiveData.postValue(entity)
//                        saveString("token", entity.access_token)
//                        //getVersion();
//                    }

            })
    }

    fun ss() {
        instance.login("admin", "123456")
            .subscribe(object : BaseObserver<UserBean>(compositeDisposable) {
                override fun onError(e: Throwable) {
                    super.onError(e)
                    ss.postValue(null)
                }

                override fun _onNext(entity: UserBean) {
                    ss.postValue(entity)
                }
//                    override fun _onNext(entity: UserBean) {
//                        userMutableLiveData.postValue(entity)
//                        saveString("token", entity.access_token)
//                        //getVersion();
//                    }
            })

    }
    //        ApiRetrofit.Companion.getInstacne().getversion().subscribe(new BaseObserver<VersionBean>(getCompositeDisposable()) {
    //            @Override
    //            public void _onNext(@NonNull VersionBean entity) {
    //                Log.e("hu",new Gson().toJson(entity));
    //                InstallApkUtils.Companion.checkVersion("http://10.81.108.65:8090/mobile/downloadFj?path="+entity.getData().getLJ(),"1.apk",false);
    //            }
    //        });
    //
    //    }

}